agent/taskresource/envFiles/envfilestatus.go

```diff
@@ -0,0 +1,63 @@
+package envFiles
+
+import (
+	resourcestatus "github.com/aws/amazon-ecs-agent/agent/taskresource/status"
+	"errors"
+	"strings"
+)
+
+type EnvironmentFileStatus resourcestatus.ResourceStatus
+
+const (
+	// zero state of a task resource
+	EnvFileStatusNone EnvironmentFileStatus = iota
+	// task resource is created
+	EnvFileCreated
```

Comment:
for my understanding - does "created" mean "downloaded and written to disk"? is there any potential for us to need to handle any inbetween states? e.g. if agent crashes after downloading but before flushing.

Reply:
yes, CREATED means downloaded and written to disk - if agent crashes while starting a task (bc here we are still creating the resoruce so task isn't started yet) then on restart the task is re-initialized anyway and the resources will be created again so there aren't really any "in between" states

Reply:
if the resource has already transitioned to CREATED and agent crashes and restarts before task moves to RUNNING, we need to handle that case by re-downloading the s3 object again..